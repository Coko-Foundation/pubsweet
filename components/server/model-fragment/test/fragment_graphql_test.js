process.env.NODE_CONFIG = `{"pubsweet":{
  "components":[
    "@pubsweet/model-user",
    "@pubsweet/model-team",
    "@pubsweet/model-fragment"
  ]
}}`

const { model: User } = require('@pubsweet/model-user')
const { dbCleaner, api } = require('pubsweet-server/test')

const { fixtures } = require('@pubsweet/model-user/test')
const authentication = require('pubsweet-server/src/authentication')

const Fragment = require('../src/fragment')

const { db } = require('@pubsweet/db-manager')

describe('Fragment queries', () => {
  let token
  let user
  let otherUser

  beforeEach(async () => {
    await dbCleaner()
    user = await new User(fixtures.user).save()
    otherUser = await new User(fixtures.otherUser).save()

    token = authentication.token.create(user)
  })

  it('lists fragments', async () => {
    await Fragment.query().insert({
      fragmentType: 'testing',
      owners: [user.id],
    })
    await Fragment.query().insert({
      fragmentType: 'testing2',
      owners: [otherUser.id],
    })

    const whereIn = jest.spyOn(db.client, 'query')

    const { body } = await api.graphql.query(
      `query {
        fragments {
          fragmentType
          owners {
            username
          }
        }
      }`,
      {},
      token,
    )

    // Gets the actual query that the database was called with
    const lastCall = whereIn.mock.calls[whereIn.mock.calls.length - 1][1]
    expect(lastCall.sql).toBe(
      'select "users".* from "users" where "id" in ($1, $2)',
    )
    expect(lastCall.bindings.sort).toBe([user.id, otherUser.id].sort)
    expect(body.data.fragments).toMatchSnapshot()
  })
})
