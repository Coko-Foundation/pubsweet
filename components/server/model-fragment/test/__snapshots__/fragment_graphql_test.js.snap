// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fragment queries lists fragments 1`] = `
Array [
  Object {
    "fragmentType": "testing",
    "owners": Array [
      Object {
        "username": "testuser",
      },
    ],
  },
  Object {
    "fragmentType": "testing2",
    "owners": Array [
      Object {
        "username": "anotheruser",
      },
    ],
  },
]
`;
